### DISCUSSION TITLE
terraform not terminated gracefully

### DISCUSSION THREAD

**Discussion by florianmutter** [suzuki-shunsuke/tfaction#1504] [Deprecated (Support Request)]
### tfaction version

latest

### Overview

When the plan or apply workflow is canceled while terraform is running the state lock is not removed.

### How to reproduce

GitHub Actions' log

```
...
module.service-account-dev["fulfill-anonymization"].google_project_iam_member.service_account["projects/puc-d-ngshop-gke-0228/roles/bigtableInstancePing"]: Refreshing state... [id=puc-d-ngshop-gke-0228/projects/puc-d-ngshop-gke-0228/roles/bigtableInstancePing/serviceAccount:fulfill-fulfill-anonymization@puc-d-ngshop-gke-0228.iam.gserviceaccount.com]
module.service-account-dev["buy-cart-processor"].google_project_iam_member.service_account["roles/secretmanager.secretAccessor"]: Refreshing state... [id=puc-d-ngshop-gke-0228/roles/secretmanager.secretAccessor/serviceAccount:buy-buy-cart-processor@puc-d-ngshop-gke-0228.iam.gserviceaccount.com]
Error: The operation was canceled.
```

### Expected behaviour

tfaction should forward the cancel signal from Github Actions to terraform.

### Actual behaviour

terraform is just killed

### Note

See discussion here about this issue: https://github.com/orgs/community/discussions/26311

Possible work around: https://github.com/orgs/community/discussions/26311#discussioncomment-7571648

**Comment by suzuki-shunsuke** [comment]
Thank you for your feedback.
I think this is the issue of GitHub Actions, but maybe we should try the workaround.

- https://github.com/orgs/community/discussions/26311#discussioncomment-7571648

**Comment by florianmutter** [comment]
Another option could be to add `if: always()` here: https://github.com/suzuki-shunsuke/tfaction/blob/main/tfmigrate-plan/action.yaml#L16 not sure what would change because of this.

This would mean it runs even if `suzuki-shunsuke/tfaction/get-target-config` would fail.
